viam_cartographer_internal_test.go

```diff
@@ -92,31 +93,44 @@ func TestPositionEndpoint(t *testing.T) {
 	var inputQuat map[string]interface{}
 
 	t.Run("empty component reference success", func(t *testing.T) {
-		svc.lidar.name = ""
+		lidarName := ""
+		mockLidar := sensors.TimedLidarSensorMock{}
```

Comment:
why create a whole lidar here when all you use is its name?

Reply:
Because the lidar is added to the `CartographerService`, where in the call to `Position` --> `CheckQuaternionFromClientAlgo` we pass the lidar name by calling `cartoSvc.lidar.Name()` (you can see this change made in viam_cartographer.go below)

Reply:
Ah I see its being placed in the svc. thanks for the clarification.